overlord/devicestate/handlers_install.go

```diff
@@ -630,6 +649,108 @@ func (m *DeviceManager) doRestartSystemToRunMode(t *state.Task, _ *tomb.Tomb) er
 	return nil
 }
 
+var seedOpen = seed.Open
+
+var maybeApplyPreseededData = func(st *state.State, ubuntuSeedDir, sysLabel, writableDir string) (preseeded bool, err error) {
+	preseedArtifact := filepath.Join(ubuntuSeedDir, "systems", sysLabel, "preseed.tgz")
+	if !osutil.FileExists(preseedArtifact) {
+		return false, nil
+	}
+
+	model, err := findModel(st)
+	if err != nil {
+		return false, fmt.Errorf("internal error: cannot find model: %v", err)
```

Comment:
I'm not sure about these "internal error" messages, I would find more useful if they gave more context information (for example, "preseed: cannot find model: %v")

Reply:
Tweaked.